src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dns/ClusterDnsCache.scala

```diff
@@ -1,90 +1,79 @@
-
-  // Responses to GetByProjectAndName message
-  sealed trait GetClusterResponse
-  case object ClusterNotFound extends GetClusterResponse
-  case object ClusterNotReady extends GetClusterResponse
-  case object ClusterPaused extends GetClusterResponse
-  case class ClusterReady(hostname: Host) extends GetClusterResponse
+  private val hostToIpMapping: ValueBox[Map[Host, IP]] = ValueBox(Map.empty)
+  def hostToIp: Map[Host, IP] = hostToIpMapping.value
+
+  sealed trait HostStatus
+  case object HostNotFound extends HostStatus
+  case object HostNotReady extends HostStatus
+  case object HostPaused extends HostStatus
+  case class HostReady(hostname: Host) extends HostStatus
 }
 
+case class DnsCacheKey(googleProject: GoogleProject, clusterName: ClusterName)
+
 /**
-  * This actor periodically queries the DB for all clusters, and updates in-memory caches of:
-  * 1. (GoogleProject, ClusterName) -> Hostname
+  * This class provides in-memory caches of:
+  * 1. (GoogleProject, ClusterName) -> HostStatus
   *    This is used by ProxyService to look up the hostname to connect to given the GoogleProject/ClusterName
   *    in the Leo request URI.
   * 2. Hostname -> IP
   *    This is used by JupyterNameService to match a "fake" hostname to a real IP address. Note
-  *    this cache is in the object not the Actor because JupyterNameService doesn't have an ExecutionContext
-  *    to query the Actor.
+  *    this cache is in the object (instead of the class) because of the way
+  *    JupyterNameService needs to access the singleton.
   * @param proxyConfig the proxy configuration
   * @param dbRef provides access to the database
   */
-class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference) extends Actor with LazyLogging {
-  var ProjectNameToHost: Map[(GoogleProject, ClusterName), GetClusterResponse] = Map.empty
-
-  import context.dispatcher
-
-  override def preStart(): Unit = {
-    super.preStart()
-    self ! RefreshFromDatabase
-  }
-
-  override def receive: Receive = {
-    case RefreshFromDatabase =>
-      queryForClusters pipeTo self
-
-    case ProcessClusters(clusters) =>
-      processClusters(clusters)
-      scheduleRefresh
-
-    // add one ready cluster to the caches now instead of waiting for the refresh cycle
-    case ProcessReadyCluster(cluster) =>
-      sender ! processReadyCluster(cluster)
-
-    case GetByProjectAndName(googleProject, clusterName) =>
-      sender ! ProjectNameToHost.getOrElse((googleProject, clusterName), ClusterNotFound)
-  }
-
-  private def scheduleRefresh = {
-    context.system.scheduler.scheduleOnce(proxyConfig.dnsPollPeriod, self, RefreshFromDatabase)
-  }
+class ClusterDnsCache(proxyConfig: ProxyConfig, dbRef: DbReference, dnsCacheConfig: ClusterDnsCacheConfig)
+                     (implicit executionContext: ExecutionContext)
+  extends LazyLogging {
+
+  def cache: LoadingCache[DnsCacheKey, Future[HostStatus]] = projectClusterToHostStatus
+
```

Comment:
Made `projectClusterToHostStatus` `private` and `def`ined getter for it so the cache isn't writable outside the class.

Reply:
But isn't it still be writable via the `cache` method? External callers could do `.cache.put(...)`.

Normally we've not exposed the cache directly at all, but defined a getter like:
```
def getHostStatus(googleProject: GoogleProject, clusterName: ClusterName): Future[HostStatus] = {
  projectClusterToHostStatus.get(DnsCacheKey(googleProject, clusterName)
}
```